src/main/java/jenkins/plugins/git/GitSCMSource.java

```diff
@@ -164,6 +164,9 @@ public GitSCMSource(String remote) {
 
     @DataBoundSetter
     public void setCredentialsId(@CheckForNull String credentialsId) {
+        if (!isFIPSCompliantTLS(credentialsId, this.remote)) {
+            LOGGER.log(Level.SEVERE, Messages.git_fips_url_notsecured());
```

Comment:
what about adding 
```
throw new IllegalArgumentException(Messages.git_fips_url_notsecured());
```
Because currently the wrong setup is still saved.
Sadly this will not be visible for the user as the output it's in the log. But hey the user have already decided too ignore the form validation so we already gave him a warning :)

Reply:
problem with this Exception is credentialsId will be null (I know, that's what we want). Anyway, I'm not sure why (maybe my machine configuration to use ssh keys?) remote are fetched without any problem if creds are null.
I'm testing it in a FIPS docker environment and rechecking if it works as expected

Reply:
Works fine in FIPS environment (seems was downloading things because of my home ssh settings).
Result in the task listener is less clear thoug
```
 > /usr/bin/git ls-remote --symref -- http://github.com/cloudbees/cloudbees-casc-items # timeout=10
ERROR: [Thu Aug 01 06:35:21 UTC 2024] Could not update folder level actions from source 3c256f49-ffdc-43a4-ad8f-f8080ba18754
[Thu Aug 01 06:35:21 UTC 2024] Finished branch indexing. Indexing took 0.68 sec
FATAL: Failed to recompute children of mbp-1
hudson.plugins.git.GitException: Command "/usr/bin/git ls-remote --symref -- http://github.com/cloudbees/cloudbees-casc-items" returned status code 128:
stdout: 
stderr: remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/cloudbees/cloudbees-casc-items/'
```
Instead of getting a custom FIPS error message we're getting invalid credentials one (expected). No strong preference on this one, maybe inclined for the custom error

Reply:
I'm on the fence with this. On the one hand, it's saved and in the other hand the logs are providing more information. 

Looking at https://github.com/search?q=org%3Ajenkinsci+GitSCMSource+language%3AJava&type=code&l=Java I'm not totally sure about what solution would be better, tbh

Reply:
As the exception is thrown in the abstract parent class, in the `retrieveActions` method, an implementation might always override it, but then the FIPS compliance is that other plugin overriding the method's responsibility.